libavutil/hwcontext_vulkan.c

```diff
@@ -1659,6 +1659,88 @@ static void vulkan_frame_free(void *opaque, uint8_t *data)
     av_free(f);
 }
 
+static int alloc_bind_single_mem(AVHWFramesContext *hwfc, AVVkFrame *f,
+                          void *alloc_pnext, size_t alloc_pnext_stride)
+{
+    int err;
+    VkResult ret;
+    AVHWDeviceContext *ctx = hwfc->device_ctx;
+    VulkanDevicePriv *p = ctx->internal->priv;
+    VulkanFunctions *vk = &p->vkfn;
+    const int planes = av_pix_fmt_count_planes(hwfc->sw_format);
+    VkBindImageMemoryInfo bind_info[AV_NUM_DATA_POINTERS] = { { 0 } };
+
+    AVVulkanDeviceContext *hwctx = ctx->hwctx;
+
+    VkImageMemoryRequirementsInfo2 req_desc = { 0 };
+    VkMemoryDedicatedAllocateInfo ded_alloc = { 0 };
+    VkMemoryDedicatedRequirements ded_req = { 0 };
+    VkMemoryRequirements2 req = { 0 };
+
+    int mem_size = 0;
+    int mem_size_list[AV_NUM_DATA_POINTERS] = { 0 };
+    int use_ded_mem;
+
+    for (int i = 0; i < planes; i++) {
+        req_desc.sType = VK_STRUCTURE_TYPE_IMAGE_MEMORY_REQUIREMENTS_INFO_2;
+        req_desc.image = f->img[i];
+
+        ded_alloc.sType = VK_STRUCTURE_TYPE_MEMORY_DEDICATED_ALLOCATE_INFO;
+        ded_alloc.pNext = (void *)(((uint8_t *)alloc_pnext) + i*alloc_pnext_stride);
+
+        ded_req.sType = VK_STRUCTURE_TYPE_MEMORY_DEDICATED_REQUIREMENTS;
+
+        req.sType = VK_STRUCTURE_TYPE_MEMORY_REQUIREMENTS_2;
+        req.pNext = &ded_req;
+
+        vk->GetImageMemoryRequirements2(hwctx->act_dev, &req_desc, &req);
+
+        if (f->tiling == VK_IMAGE_TILING_LINEAR)
+            req.memoryRequirements.size = FFALIGN(req.memoryRequirements.size,
+                                                  p->props.properties.limits.minMemoryMapAlignment);
+
+        mem_size_list[i] += FFALIGN(req.memoryRequirements.size/hwfc->width, 64) * hwfc->width;
```

Comment:
Where is the 64 come from? in the future platforms it may change.
Could we get the 64 from somewhere?

Reply:
Here I made a mistake, I want the offset to be multiple of width.